frontend/src/hooks/useAsyncError.ts

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+/**
+ * allow an async action to throw a regular error that will be caught be our react error boundary. example:
+ * const throwError useAsyncError();
+ * try{
+ *  await someAsyncFunction();
+ * } catch(error) {
+ *  throwError(error);
+ * }
+ */
+export const useAsyncError = () => {
```

Comment:
Note, this should only be used on errors that leave the page in a non-usable. Ie. if the property list view page fails to load property data, or similar.

Reply:
Is `Async` relevant?  Seems to me that it's just throwing an `any` error.
I also don't understand the use of the state here?  It looks like you're returning a function that calls the set state, which throws the error.

Reply:
I'll link you to the article that I got this from, probably do a better job of explaining.

Reply:
https://medium.com/trabe/catching-asynchronous-errors-in-react-using-error-boundaries-5e8a5fd7b971

Reply:
in summary, the only way to rethrow an async action in a manner that an error boundary will catch is by converting an async error to a regular js error. How do you do this? You set a state var in an async function. This causes a re-render in regular, non-async react js. We then use a useEffect such that when the error state var is set we just throw the error